src/mongoc/mongoc-client.c

```diff
@@ -1783,8 +1783,20 @@ _mongoc_client_command_with_opts (mongoc_client_t *client,
       if ((mode & MONGOC_CMD_WRITE) &&
           !mongoc_write_concern_is_default (default_wc) &&
           (!opts || !bson_has_field (opts, "writeConcern"))) {
-         bool is_fam =
-            !strcasecmp (_mongoc_get_command_name (command), "findandmodify");
+         const char *command_name;
+         bool is_fam;
+
+         command_name = _mongoc_get_command_name (command);
+
+         if (!command_name) {
+            bson_set_error (error,
+                            MONGOC_ERROR_COMMAND,
+                            MONGOC_ERROR_COMMAND_INVALID_ARG,
+                            "Empty command document");
+            GOTO (err);
+         }
+
+         is_fam = !strcasecmp (command_name, "findandmodify");
```

Comment:
We have bson_strcasecmp, let's standardize on it.

Reply:
Considering that `strcasecmp()` isn't even defined on Windows, I'm not sure how this code would have even compiled. There are many more uses of `strcasecmp()` throughout libmongoc as well.

I ultimately traced it back to this bit in `mongoc-util-private.h`, which is commonly included:

```
/* string comparison functions for Windows */
#ifdef _WIN32
#define strcasecmp _stricmp
#define strncasecmp _strnicmp
#endif
```

I'm not going to change any of this in this PR, but I'll create a ticket so that you can standardize down the line. IMO, the `mongoc-util-private.h` approach is preferable to adding a custom function that delegates. Even if that function is inlined by the compiler, there is cognitive overhead with having to use `bson_strcasecmp()` for case-insensitive but `strcmp()` for regular comparisons.

Reply:
Fair.

Reply:
Opened [CDRIVER-2437](https://jira.mongodb.org/browse/CDRIVER-2437).